codice/alliance
# CAL-312 Upgraded Karaf to 4.1.1

# DO NOT MERGE
TODO
- Blocked by https://github.com/codice/ddf/pull/2061
- Manual testing
#### What does this PR do?
This PR upgrades Alliance to Karaf 4.1.1.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@clockard @vinamartin @ryeats @brendan-hofmann @bdeining @paouelle 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@clockard
@coyotesqrl
@kcwire
@figliold
@lessarderic
@pklinef
@stustison
#### How should this be tested?
- Test that a full build succeeds.
- Drop `.cfg` and `.config` files in the `/etc` directory, and confirm that the configurations are installed successfully.
- Manually test basic functionality of Alliance.
- Confirm that the `./bin` scripts work as expected.
- Walkthrough the Hardening Guide.
- Confirm that the `waitForReady` command still works.
- Confirm that the `migration:export` command and exporting configurations from the Admin Console still work.
- Confirm that installing the `profile-development` works.
- Confirm that logging still works and is still configurable from `org.ops4j.pax.logging.cfg` and from the Karaf console. Also, try switching the logging configuration to the xml-style configuration (see comment in `org.ops4j.pax.logging.cfg`, and confirm that logging still works and is configurable from `log4j2.config.xml` and from the Karaf console.
#### Any background context you want to provide?
#### What are the relevant tickets?
[CAL-312](https://codice.atlassian.net/browse/CAL-312)
[CAL-289](https://codice.atlassian.net/browse/CAL-289)

NOTE: See https://github.com/codice/ddf/pull/2061 for generated tickets and known issues.
#### Screenshots (if appropriate)
#### Checklist:
